### DISCUSSION TITLE
Build(time) of Krokiet very high. (?)

### DISCUSSION THREAD

**Discussion by fabiscafe** [qarmin/czkawka#1220] [General]
Hi. I'm the maintainer of the Czkawka AUR PKGBUILD on Archlinux. Currently I also include Krokiet in the build. Since this change the time to build everything is >10 times the time of <7.0.0.

It looks like Krokiet only utilizes one thread of my CPU, while everything else uses my full CPU to build . Is this a limitation of a component or maybe a problem on my side?

My build phase looks like this:
```
build() {
  cd ${pkgbase}

  # Keep rust/cargo build-dependency management inside the build directory
  export CARGO_HOME="${srcdir}/cargo"

  cargo build \
    --bin czkawka_cli \
    --bin czkawka_gui \
    --bin krokiet \
    --features heif \
    --release --verbose
}
```
I dont want to open an issue, if the problem is on my side. Thats why I'm here :smile: 

 This is with cargo+rustc 1.76.0

**Comment by qarmin** [comment]
This is caused by rustc, which wants to optimize krokiet, but sroa optimization is not designed to work with slint specific, auto-generated ~90000 lines of rust code

https://github.com/rust-lang/rust/issues/121354
https://github.com/qarmin/czkawka/issues/1218

**Comment by fabiscafe** [comment]
Ahh Thank you! That's helpful :+1:
